plutus-core/testlib/PlutusCore/Generators/Internal/Builtin.hs

```diff
@@ -32,6 +34,27 @@ import Hedgehog.Gen qualified as Gen
 import Hedgehog.Range qualified as Range
 import Type.Reflection
 
+class GenTypedTerm uni where
```

Comment:
Arguably these belong in `PlutusCore.Generators.AST` since they both generate `Term`s?

Reply:
Also, Haddock. Even something simple like: "This class exists so we can provide an ad-hoc typed term generator for various universes. This is difficult to implement generically since we usually rely-on a universe-specific generator for well-typed constants within that universe."

Reply:
> Arguably these belong in PlutusCore.Generators.AST since they both generate Terms?

I added a TODO here. Ideally we should merge the `genConstant` here with `PlutusIR.Generators.AST.genConstant`, since they are both supposed to generate constants in `DefaultUni`. But currently doing so causes some test failures (e.g., parser roundtrip tests), which needs to be investigated.